src/inc/internal/GeneralUtil.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once

```

Comment:
This file was meant to be a helper for callers to the dll. Internally we shouldnt use it and use std::string, we usually implement internal interface to return std::string when needed.

Reply:
I added this file, which has helper functions needed to process the input in the PackBundle method. It does version conversion to and from uint64, checks if output file is already present or if the output file is a directory. Do you mean I should not be adding this file?

Reply:
Talk offline about this. The goal should be to never use an RAII wrapper for strings as it requires marshaling out a string and inside the dll boundary that's not the best approach because of performance. Also this class assumes the factory object uses std::malloc  which might not be the case for some windows callers. 

As for ConvertVersionStringToUint64 ConvertVersionToString I think we can rename the file to VersionHelpers.hpp/cpp to reflect more the nature of these methods or perhaps add them in StringHelper.hpp and create a StringHelper.cpp

I think that GeneralUtil is just to board and anything can live here, so separation will make the code more easy to read in the future.

---
In reply to: [517406849](https://github.com/microsoft/msix-packaging/pull/389#discussion_r517406849) [](ancestors = 517406849)

Reply:
Removed the extra TEXT structure that was being used. Created IAppxTargetDeviceFamilyInternal and IAppxQualifiedResourcesInternal instead to fetch these string values. Renamed the GeneralUtil file to VersionHelpers. I will create function specific file names going forward.